azdev/operations/linter/rule_decorators.py

```diff
@@ -4,54 +4,81 @@
 # license information.
 # -----------------------------------------------------------------------------
 
-from .linter import RuleError
+from knack.util import CLIError
+from .linter import RuleError, LinterSeverity
 
 
-def help_file_entry_rule(func):
-    return _get_decorator(func, 'help_file_entries', 'Help-Entry: `{}`')
+class AbstractRule(object):
```

Comment:
I prefer not name it `Abstract` because it did't give any constraint to the subclass, it doesn't have any `@abstractmethod` decorator to the instance method.
How about 
```suggestion
class BaseRule:
```